python/plugins/processing/core/parameters.py

```diff
@@ -434,14 +441,19 @@ def getAsScriptCode(self):
 class ParameterNumber(Parameter):
 
     def __init__(self, name='', description='', minValue=None, maxValue=None,
-                 default=0.0, optional=False):
-        Parameter.__init__(self, name, description, optional)
-        try:
-            self.default = int(unicode(default))
-            self.isInteger = True
-        except:
-            self.default = float(default)
+                 default=None, optional=False):
```

Comment:
I fact, I apply to default param the default value.

If you compare the code, we have:
```python
self.value = None
```
and now we have:
```python
default=None
```
because we have:
```python
self.value = default
```
The idea of this code is also to discuss about the default value of parameter default value.

Param | default value now | default value before | value in init before
--------- | ------------------------- | ------------------------ | --------------------------
Boolean | True | True | None
Crs | None | 'EPSG:4326' | None
Extent | None | '0,1,0,1' | None
File | '' | No default | None
FixedTable | None | No default | None
MultipleInput | None | No default | None
Number | None | 0.0 | None
Range | None | '0,1' | None
Raster | None | No default | None
Selection | 0 | 0 | None
String | '' | '' | None
Table | None | No default | None
TableField | None | No default | None
Vector | None | No default | None
GeometryPredicate |  [] | [] | None

For me Except for Boolean or Selection, for which None has to be translated to Boolean, I'm for default to None.